packages/errors/README.md

```diff
@@ -46,6 +79,43 @@ OperationalError.isErrorMarkedAsOperational(new OperationalError('example messag
 OperationalError.isErrorMarkedAsOperational(new Error('example message')); // false
 ```
 
+### `HttpError`
+
+The `HttpError` class extends `OperationalError` and represents an HTTP error status. It can work in the same way as a normal error, expecting a message. In this case it will represent an HTTP `500`:
+
+```js
+throw new HttpError('example message');
+```
+
+You can alternatively construct an HTTP error with a data object. This accepts a `statusCode` property, which is a valid HTTP status code number, as well as all of the properties you can set in [`OperationalError`](#operationalerror):
+
+```js
+throw new HttpError({
+    message: 'your thing was not found',
+    statusCode: 404
+});
+```
+
+It's also possible to create an HTTP error with a status code alone, which will default the message to the corresponding HTTP status message:
+
+```js
+throw new HttpError(404);
+```
+
+Error properties can be accessed like any other property:
+
+```js
+error.message // your thing was not found
+error.statusCode // 404
+error.status // 404
+error.statusMessage // Not Found
+error.code // HTTP_404
+```
+
+#### Why use this over `http-errors`?
+
+The benefit of using this error rather than the excellent [http-errors](https://github.com/jshttp/http-errors#readme) library is that we extend `OperationalError` by default. This means that all HTTP errors you throw are considered "known errors" by the rest of our tooling. We also set a `code` property by default which results in less code in our monitoring dashboards – we don't need to check both `code` and `statusCode` properties to determine the type of error thrown.
```

Comment:
**suggestion that would be a lot of work so probably just ignore me but might be worth a thought**: if you were actually using Typescript, `OperationalError` could become an interface, and you could do something like

```ts
import HttpError from 'http-errors'
class OperationalHttpError extends HttpError implements OperationalError {}
```

Reply:
I appreciate the suggestion but yeah think I don't want to go wholesale TypeScript on this. Feels like a big commitment to a build step and change of language for the purpose of including a third-party lib. Also, in this scenario, we'd have to reimplement the things currently on the `OperationalError` class right, Like the `data`, `isOperational`, and `code` properties? Or possibly I'm misunderstanding interfaces in TS?

Reply:
well, it would be a separate interface that both `HttpError` and `OperationalError` implement. just a way of structuring that for some kind of typed inheritance. irrelevant tho